src/log/log.c

```diff
@@ -2537,7 +2537,7 @@ __wt_log_scan(WT_SESSION_IMPL *session, WT_LSN *lsnp, uint32_t flags,
 				 * must be salvaged.
 				 */
 				need_salvage = true;
-				WT_ERR(__log_salvage_message(session,
+				WT_TRET(__log_salvage_message(session,
```

Comment:
@sueloverso it's not obvious but `WT_TRET` doesn't return or goto error, it just sets up `ret` - is that what you want here? Or should there be a subsequent `WT_ERR` call?

Reply:
This code is in the "checksums don't match" area. After the various salvage calls, there is an unconditional `break`.

Reply:
@sueloverso, I think the change is correct.  But I agree with Alex that it's hard to reason about (an unusual usage of `WT_TRET`).  Perhaps clearer would be keep `WT_TRET`, but add `goto truncate`, with the new label just after the loop?  And there's another occurrence of the same a few lines away.

Reply:
Got it - do we need to add a `WT_ERR(ret);` immediately [after the](https://github.com/wiredtiger/wiredtiger/pull/4339/files#diff-764f5432bdeaa6ffb0296d56d50ca6ceR2599) loop then?